test/commands/mentor/discussion/finish_abandoned_test.rb

```diff
@@ -16,74 +16,23 @@ class Mentor::Discussion::FinishAbandonedTest < ActiveSupport::TestCase
       end
     end
 
-    test "reputation awarded when abandoning from status #{status}" do
-      finished_at = Time.current.utc - 1.week
-      discussion = create(:mentor_discussion, status:, finished_at:, updated_at: finished_at)
-
-      User::ReputationToken::Create.expects(:defer).with(
-        discussion.mentor,
-        :mentored,
-        discussion:
-      )
-
-      Mentor::Discussion::FinishAbandoned.()
-    end
-
-    test "awards mentor badge when abandoning from status #{status}" do
-      mentor = create :user
-      9.times do |_idx|
-        create :mentor_discussion, :student_finished, mentor:
-      end
-
-      finished_at = Time.current.utc - 1.week
-      create(:mentor_discussion, status:, mentor:, finished_at:, updated_at: finished_at)
-      refute mentor.badges.present?
-
-      perform_enqueued_jobs do
-        Mentor::Discussion::FinishAbandoned.()
-      end
-
-      assert_includes mentor.reload.badges.map(&:class), Badges::MentorBadge
-    end
-
-    test "awards mentored trophy when abandoning from status #{status}" do
-      student = create :user
-      finished_at = Time.current.utc - 1.week
-      request = create(:mentor_request, student:)
-      create(:mentor_discussion, status:, student:, finished_at:, updated_at: finished_at, request:)
-      refute student.badges.present?
-
-      perform_enqueued_jobs do
-        Mentor::Discussion::FinishAbandoned.()
-      end
-
-      assert_includes student.reload.trophies.map(&:class), Track::Trophies::MentoredTrophy
-    end
-
-    test "adds metric when abandoning from status #{status}" do
-      student = create :user
-      finished_at = Time.current.utc - 1.week
-      discussion = create(:mentor_discussion, status:, student:, finished_at:, updated_at: finished_at)
```

Comment:
All of this got moved into a dedicated test for the `ProcessFinished` command to DRY things up in the tests.